### PR TITLE
add tests for resmap

### PR DESCRIPTION
**Pull request by ryancox**
This patch brings tests for the resmap package to 90%+

```
ok  	sigs.k8s.io/kustomize/pkg/resmap	1.071s	coverage: 90.5% of statements
```

### CODE REVIEW

File: pkg/resmap/resmap.go [hunk 1]
```diff
@@ -125,20 +123,6 @@ func (m ResMap) DeepCopy() ResMap {
 	return mcopy
 }
 
-func (m ResMap) insert(newName string, obj *unstructured.Unstructured) error {
```
Comment (ryancox) [inline pkg/resmap/resmap.go]: this code is dead AFAICT

**Comment by Liujingfang1** [comment]
@ryancox Thank you for adding those tests. I'll take a look

File: pkg/resmap/resmap_test.go [hunk 2]
```diff
@@ -70,6 +70,202 @@ metadata:
...
+			map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "cm2",
+				},
+			}),
+	}
+
+	rm2 := rm1.DeepCopy()
```
Comment (Liujingfang1) [inline pkg/resmap/resmap_test.go]: Can you also make sure the resources in `rm1` and `rm2` are different objects?

Comment (ryancox) [reply pkg/resmap/resmap_test.go]: done

**Pull request merged**